torch/csrc/CudaIPCTypes.h

```diff
@@ -0,0 +1,162 @@
+#pragma once
+#ifdef USE_CUDA
+#include <c10/core/Allocator.h>
+#include <c10/cuda/CUDACachingAllocator.h>
+#include <c10/cuda/CUDAException.h>
+#include <c10/cuda/CUDAGuard.h>
+#include <c10/cuda/CUDAStream.h>
+#include <c10/util/Logging.h>
+#include <cuda_runtime_api.h>
+#include <iostream>
+
+namespace torch {
+
+constexpr uint CUDA_IPC_REF_COUNTER_FILE_SIZE = 10000;
+constexpr uint CUDA_IPC_WARN_AFTER_X_BLOCKS_IN_LIMBO = 1000;
+
+void CudaIPCCollect();
+
+at::DataPtr GetNewRefCountedSentData(void* data, at::Device device);
```

Comment:
Documentation for this function would be especially helpful. (Also see internal usage note above.)